pkg/operator/profilecalculator.go

```diff
@@ -626,6 +630,46 @@ func (pc *ProfileCalculator) getNodePoolNameForNode(node *corev1.Node) (string,
 	return nodePoolName, nil
 }
 
+// TunedRecommend returns a name-sorted TunedProfile slice out of
+// a slice of Tuned objects.
+func TunedProfiles(tunedSlice []*tunedv1.Tuned) []tunedv1.TunedProfile {
+	tunedProfiles := []tunedv1.TunedProfile{}
+	m := map[string]tunedv1.TunedProfile{}
+
+	for _, tuned := range tunedSlice {
+		if tuned.Spec.Profile != nil {
+			for _, v := range tuned.Spec.Profile {
+				if v.Name != nil && v.Data != nil {
+					if existingProfile, found := m[*v.Name]; found {
+						if *v.Data == *existingProfile.Data {
+							klog.Infof("duplicate profiles names %s but they have the same contents", *v.Name)
+						} else {
+							klog.Errorf("ERROR: duplicate profiles named %s with different contents found in Tuned CR %q", *v.Name, tuned.Name)
+						}
+					}
+					m[*v.Name] = v
+				}
+			}
+		}
+	}
```

Comment:
I know this is a code movement, but keeping the happy path on the left usually improves the readability:
```
	for _, tuned := range tunedSlice {
		if tuned.Spec.Profile == nil {
			continue
		}
		for _, v := range tuned.Spec.Profile {
			if v.Name == nil || v.Data == nil {
				continue
			}
			if existingProfile, found := m[*v.Name]; found {
				if *v.Data == *existingProfile.Data {
					klog.Infof("duplicate profiles names %s but they have the same contents", *v.Name)
				} else {
					klog.Errorf("ERROR: duplicate profiles named %s with different contents found in Tuned CR %q", *v.Name, tuned.Name)
				}
			}
			m[*v.Name] = v
		}
	}
```

Reply:
I agree.  I usually do this, not sure why I didn't in this case.  Will change this.  Thank you.